File: Frontend/src/types/studySession.ts
export interface StudyTask {
  task_id: number;
  roadmap_id: number;
  module_id: number;
  topic_id: number;
  title: string;
  topic: string;
  module: string;
  estimated_minutes: number;
  status: string;
}

export interface StudySession {
  requested_minutes: number;
  total_minutes: number;
  task_count: number;
  tasks: StudyTask[];
}

File: Backend/app/schemas/study_session.py
from pydantic import BaseModel


class StudyTask(BaseModel):
    task_id: int
    roadmap_id: int
    module_id: int
    topic_id: int

    title: str
    topic: str
    module: str

    estimated_minutes: int
    status: str


class StudySessionResponse(BaseModel):
    requested_minutes: int
    total_minutes: int
    task_count: int
    tasks: list[StudyTask]